### ISSUE TITLE
Using GGally::ggpairs more in Describe menu?

### ISSUE THREAD

**Issue by dannyparsons** [IDEMSInternational/R-Instat#3942]
I'm looking into the correlations dialog as it's going to be useful in the procurement menu to compare red flag columns. Some of the red flag columns are binary i.e. just 0 or 1. In R this could be represented as a numeric column of 1s and 0s or as logical type column of `TRUE`s and `FALSE`s.

I first noticed that the dialog only allows numeric columns, so logical type columns couldn't be used. But testing all the functions used in the dialog, logical columns can be treated as numeric (sort of). This links to #3925 on how our selectors could exclude types instead of including them, so that logical is considered "numeric".

The `GGally::ggpairs` function does give slightly different outputs for logical columns compared to numeric columns.
This is with `a` and `b` as numeric 0-1 columns:
![image](https://user-images.githubusercontent.com/11214914/30598957-08f8d27a-9d53-11e7-9382-ad250820e01b.png)
And this is with `a` and `b` as logical columns:
![image](https://user-images.githubusercontent.com/11214914/30598984-234b96bc-9d53-11e7-90e0-dc03b076bdcb.png)

So slightly strange graphs on the diagonal for logical columns but otherwise the same.

I also tried `GGally::ggpairs` with factor columns just to see and it gives an interesting output:
![image](https://user-images.githubusercontent.com/11214914/30599174-94c1ddc4-9d53-11e7-897d-178e89aadb92.png)

So two questions:

1. Should we allow logical columns in the correlations dialog? I feel we probably should because I think we want to encourage binary columns to be stored as logical types instead of 0/1 numeric columns but if we don't allow logical here then users would have to convert. And there are probably other numeric-like columns which could work but are currently excluded.

2. Should we be making more use of `GGally::ggpairs` since it seems more general than correlations? The graphs part are sort of what we have under the two variable graph dialog. Could we include this in that dialog, or have a separate item under Describe > Two Variables since it's a table/graph mix?

**Comment by dannyparsons** [comment]
There is also `GGally::ggscatmat` which is similar but strictly for numeric columns:

`GGally::ggscatmat(survey)
`![image](https://user-images.githubusercontent.com/11214914/30694281-3b49486c-9ecb-11e7-99ab-333086209a85.png)

Slightly off topic, but it also has other interesting functions such as `GGally::ggmatrix` which is similar to `gridExtra::grid.arrange` but produces a `gg` (not `ggplot`) object. This might have some advantages. Worth a general investigation of this package.

**Comment by rdstern** [comment]
There is one urgent task to correct in the correlations dialogue.  The use of ggpairs is in the first tutorial and it has changed from what it used to do.  It no longer gives the sort of graph seen just above.  Instead it repeats the lower triangle above as well.  That's very simple to change.  Currently the command generated is as follows:

GGally::ggpairs(data=diamonds, columns=c("carat","depth","y","z"), **upper=list(continuous=GGally::wrap('points')),** lower=list(continuous=GGally::wrap('points')))

If the upper argument is simply deleted, then we should get back to what we had before.  At the same time we could perhaps allow logical columns as Danny suggests above.

We then keep this issue open, because of the more general points on ggpairs.  I assume they could go conveniently in our Describe > Two Variables menu.  There is a lot to do in that menu.  Once this point is changed, I suggest we change the milestone to 5.1.

**Comment by Muthenya** [comment]
@rdstern the logical columns are allowed. The change was already made, i.e, for the **Multiple Columns**.

**Comment by dannyparsons** [comment]
This small change suggestion would have been better as a separate issue instead of added here because it meant that this issue was closed by @Muthenya's pull request and I had to reopen manually. We also wouldn't have to think about changing milestones if this was separate. For our workflow having many small issues is much easier to work with because it makes linking to pull requests easier.

**Comment by dannyparsons** [comment]
Yes, I also combined two separate points in one issue here. The correlations dialog part is resolved and I've renamed the issue.

**Comment by dannyparsons** [comment]
We could still fit this in with our current Two Variable dialog setup by producing an n x 1 table of graphs?
